### PR TITLE
Prevent map refocus if control click event is from the keyboard

### PR DESCRIPTION
**Pull request by Jakobud**
Prevent map refocus if control click event is from the keyboard, which is determined/assumed by the click X,Y being at 0,0. There is definitely an assumption there that it is extremely unlikely that you would be clicking a control with your mouse at 0,0 (at the very top left of the screen).

Not sure if this is the best solution, but it does work.

### CODE REVIEW

**Comment by Jakobud** [comment]
for #3159

**Comment by Jakobud** [comment]
Errors galore in the build! Not sure how to fix those :-P

File: src/control/Control.js [hunk 1]
```diff
@@ -71,8 +71,9 @@ L.Control = L.Class.extend({
 		return this;
 	},
 
-	_refocusOnMap: function () {
-		if (this._map) {
+	_refocusOnMap: function (e) {
+		// if map exists and event is keyboard event
```
Comment (mourner) [inline src/control/Control.js]: you probably meant "is not a keyboard event" :)

File: src/control/Control.js [hunk 2]
```diff
@@ -71,8 +71,9 @@ L.Control = L.Class.extend({
 		return this;
 	},
 
-	_refocusOnMap: function () {
-		if (this._map) {
+	_refocusOnMap: function (e) {
+		// if map exists and event is keyboard event
+		if (this._map && (e.x > 0 && e.y > 0 && e.screenX > 0 && e.screenY > 0)) {
```
Comment (mourner) [inline src/control/Control.js]: Try adding a check for truthy `e`, maybe it'll fix tests.

**Comment by mourner** [comment]
Yeah, I don't know whether there's a more reliable way to detect this. Bummer... I think this will be good for now.

**Comment by mourner** [comment]
:+1:

**Pull request merged**